src/gei/Commands/WaitForMigrationCommand.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using OctoshiftCLI.Extensions;
+
+[assembly: InternalsVisibleTo("OctoshiftCLI.Tests")]
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class WaitForMigrationCommand : Command
+    {
+        internal int WaitIntervalInSeconds = 10;
+
+        private readonly OctoLogger _log;
+        private readonly ITargetGithubApiFactory _targetGithubApiFactory;
+
+        public WaitForMigrationCommand(OctoLogger log, ITargetGithubApiFactory targetGithubApiFactory) : base("wait-for-migration")
+        {
+            _log = log;
+            _targetGithubApiFactory = targetGithubApiFactory;
+
+            Description = "Waits for migration(s) to finish and reports all in progress and queued ones.";
+            Description += Environment.NewLine;
+            Description += "Note: Expects GH_PAT env variables to be set.";
+
+            var githubOrg = new Option<string>("--github-org") { IsRequired = true };
+            var migrationId = new Option<string>("--migration-id")
+            {
+                IsRequired = false,
+                Description = "Waits for the specified migration to finish."
+            };
+            var verbose = new Option("--verbose") { IsRequired = false };
+
+            AddOption(githubOrg);
+            AddOption(migrationId);
+            AddOption(verbose);
+
+            Handler = CommandHandler.Create<string, string, bool>(Invoke);
+        }
+
+        public async Task Invoke(string githubOrg, string migrationId = null, bool verbose = false)
+        {
+            _log.Verbose = verbose;
+
+            var hasMigrationId = !migrationId.IsNullOrWhiteSpace();
```

Comment:
why was the codeql warning about migrationId possibly being null here closed? that sounds like a legit problem. Won't this crash if migrationId is null?

Reply:
It's not legit since `IsNullOrWhiteSpace()` is an extension method that I wrote so if the `migrationId` is `null` it returns null.